viewer-prototype/src/browser/trace-viewer/components/abstract-tree-output-component.tsx

```diff
@@ -9,19 +9,24 @@ export abstract class AbstractTreeOutputComponent<P extends AbstractOutputProps,
     renderMainArea(): React.ReactNode {
         const treeWidth = this.props.style.width - this.props.style.chartWidth - this.getHandleWidth();
         return <React.Fragment>
-            <div className='output-component-tree' style={{ width: treeWidth, height: this.props.style.height }}>
+            <div ref={this.treeRef} className='output-component-tree' onScroll={ev => { this.ScrollSync() }} style={{ width: treeWidth, height: this.props.style.height }}>
                 {this.renderTree()}
             </div>
-            <div className='output-component-chart' style={{ width: this.props.style.chartWidth, backgroundColor: '#3f3f3f' }}>
+            <div ref={this.chartRef} className='output-component-chart' style={{ width: this.props.style.chartWidth, backgroundColor: '#3f3f3f' }}>
                 {this.renderChart()}
             </div>
         </React.Fragment>;
     }
 
+    treeRef = React.createRef<any>();
+    chartRef = React.createRef<any>();
+
     abstract renderTree(): React.ReactNode;
 
     abstract renderChart(): React.ReactNode;
 
+    abstract ScrollSync(): void;
```

Comment:
Can you not make it abstract and just override it in the timegraph? Instead of requiring to implement an empty function in all sub-classes.

Also, to make clearer that this method is called when the tree is scrolled, not the chart, you could call it treeScrolled() (start the method name with lower case)